test-suite/idl-schema/properties/properties.json

```diff
@@ -0,0 +1,90 @@
+[
+  {
+    "description": "Should have accurate property types",
+    "schemas": [
+      {
+        "$schema": "https://json-schema.org/draft/2020-12/idl-schema",
+        "name": "SomeTitle",
+        "type": "object",
+        "properties": {
+          "StringProperty": {
+            "type": "string"
+          },
+          "BooleanProperty": {
+            "type": "boolean"
+          },
+          "NumberProperty": {
+            "type": "number"
+          },
+          "NullProperty": {
+            "type": "null"
+          }
+        }
+      }
+    ],
+    "tests": [
+      {
+        "assertion": "hasClass",
+        "tests": [
+          {
+            "assertion": "hasName",
+            "arguments": [
+              "SomeTitle"
+            ]
+          },
+          {
+            "assertion": "hasProperty",
+            "tests": [
+              {
+                "assertion": "hasName",
+                "arguments": [
+                  "StringProperty"
+                ]
+              },
```

Comment:
This should be an argument to the "hasProperty" assertion, not an assertion itself. Once you've determined that there exists a property with this name, then you can do additional tests on the property that was found ("tests") such as asserting it's type.

Reply:
Right. Most assertions should have "arguments". We can't check if a type has a property if we don't tell it the name of the property check for.

Reply:
Yea, good point, changing.

Reply:
The same thing applies to "hasClass". That needs to be updated as well.

Reply:
Good catch, changed 🙂